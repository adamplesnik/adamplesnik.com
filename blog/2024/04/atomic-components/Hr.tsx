import { addWithSpace } from '@/utils/addWithSpace'
import { HTMLAttributes } from 'react'

const Hr = ({ text, className, ...rest }: HrProps) => {
  return (
    <div
      className={
        'my-4 flex items-center gap-2 text-zinc-700 dark:text-zinc-300' + addWithSpace(className)
      }
      {...rest}
    >
      {text && <span className="text-xs font-medium">{text}</span>}
      <hr className="h-px flex-1 border-none bg-current" />
    </div>
  )
}

export type HrProps = {
  text?: string
} & HTMLAttributes<HTMLDivElement>

export default Hr
