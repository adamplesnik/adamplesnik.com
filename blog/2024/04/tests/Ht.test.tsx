import renderer from 'react-test-renderer'
import { expect, test } from 'vitest'
import Hr from './Hr'
import { toJson } from './utils'

test('Add class name to the component', () => {
  const component = renderer.create(<Hr className="test-class" />)
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Render simple separator', () => {
  const component = renderer.create(<Hr />)
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Render separator with text in the beginning', () => {
  const component = renderer.create(<Hr text="We begin here" />)
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Render separator with text at the end', () => {
  const component = renderer.create(<Hr text="We end there" textOrientation="end" />)
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
