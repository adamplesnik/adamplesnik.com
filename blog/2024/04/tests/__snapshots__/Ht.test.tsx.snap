// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add class name to the component 1`] = `
<div
  className="my-4 flex items-center gap-2 text-gray-700 dark:text-gray-300 test-class"
>
  <hr
    className="h-px flex-1 border-none bg-current"
  />
</div>
`;

exports[`Render separator with text at the end 1`] = `
<div
  className="my-4 flex items-center gap-2 text-gray-700 dark:text-gray-300"
>
  <span
    className="text-xs font-medium order-last"
  >
    We end there
  </span>
  <hr
    className="h-px flex-1 border-none bg-current"
  />
</div>
`;

exports[`Render separator with text in the beginning 1`] = `
<div
  className="my-4 flex items-center gap-2 text-gray-700 dark:text-gray-300"
>
  <span
    className="text-xs font-mediumfalse"
  >
    We begin here
  </span>
  <hr
    className="h-px flex-1 border-none bg-current"
  />
</div>
`;

exports[`Render simple separator 1`] = `
<div
  className="my-4 flex items-center gap-2 text-gray-700 dark:text-gray-300"
>
  <hr
    className="h-px flex-1 border-none bg-current"
  />
</div>
`;
